synapse/rest/client/v2_alpha/register.py

```diff
@@ -457,6 +461,11 @@ async def on_POST(self, request):
             )
 
             return 200, result
+        elif self.auth.has_access_token(request):
+            raise SynapseError(
+                400,
+                "A type of m.login.application_service must be provided when registering as an appservice",
+            )
```

Comment:
I don't think there is a case currently where you can register with an access token without being an appservice?

Reply:
The reason for keeping this `elif` around is that previously we used the access token to determine if the request was on the behalf of an appservice. Rather than throwing that away I wanted to throw an explicit error for people running non-compliant appservices (hence the very specific error message).

If we throw this statement away, the token will be ignored and requests that previously successfully created an appservice user may either fail with a somewhat unhelpful error (like "you didn't specify a password"), or may succeed in creating a non-appservice user which would probably do more harm than good.

Reply:
the point is more that the error message is ... unclear.

Reply:
suggested a rephrasing.